strawberryfields/api/result.py

```diff
@@ -138,6 +138,17 @@ def __repr__(self):
         if self.samples.ndim == 2:
             # if samples has dim 2, assume they're from a standard Program
             shots, modes = self.samples.shape
+
+            if self.ancilla_samples is not None:
+                ancilla_modes = 0
+                for i in self.ancilla_samples.keys():
+                    ancilla_modes += len(self.ancilla_samples[i])
+                return (
+                    "<Result: shots={}, num_modes={}, num_ancillae={}, contains state={}>".format(
+                        shots, modes, ancilla_modes, self._is_stateful
```

Comment:
Worth checking if this case (and the suggested update) works well for `Result` by a small unit test.
```suggestion
                    f"<Result: shots={shots}, num_modes={modes}, num_ancillae={ancilla_modes},"
                    f"contains state={self._is_stateful}>"
```

Reply:
I've updated it, and I already snuck in that unit test in line 145 of `tests/integration/test_engine_integration.py` 😎